submissions/krylenger/friends/app.js

```diff
@@ -0,0 +1,259 @@
+  }
+  if (filtersState.sortName === 'name_AZ') {
+    filteredUsersData = sortByParam(filteredUsersData, 'name', 'ascending');
+  }
+  if (filtersState.sortName === 'name_ZA') {
+    filteredUsersData = sortByParam(filteredUsersData, 'name', 'descending');
+  }
+  if (filtersState.sortAge === 'age_19') {
+    filteredUsersData = sortByParam(filteredUsersData, 'age', 'ascending');
+  }
+  if (filtersState.sortAge === 'age_91') {
+    filteredUsersData = sortByParam(filteredUsersData, 'age', 'descending');
+  }
+  if (filtersState.sortRegistrationDate === 'registrationDate_NEW') {
+    filteredUsersData = sortByParam(filteredUsersData, 'registrationDate', 'ascending');
+  }
+  if (filtersState.sortRegistrationDate === 'registrationDate_OLD') {
+    filteredUsersData = sortByParam(filteredUsersData, 'registrationDate', 'descending');
+  } 
+
+  renderUserCards(filteredUsersData);
+}
+
+const addEventListeners = (usersData) => {
+
+  search.addEventListener('keyup', ({target}) => {
+    filtersState.search = target.value;
+    filterCards(usersData);
+  })
+
+  asideFilterMale.addEventListener("click", ({ target }) => {
+    filtersState.gender = target.id;
+		filterCards(usersData);
+  });
+  
+  asideFilterFemale.addEventListener("click", ({ target }) => {
+		filtersState.gender = target.id;
+		filterCards(usersData);
+  });
+
+  asideFilterCountry.addEventListener("change", ({ target }) => {
+		filtersState.country = target.value;
+		filterCards(usersData);
+  });
+
+  asideFilterAge.addEventListener("change", ({ target }) => {
+		filtersState.age = target.value;
+		filterCards(usersData);
+  });
+
+	buttonSortAZ.addEventListener("click", ({ target }) => {
+		filtersState.sortName = target.id;
+		filtersState.sortAge = null;
+		filtersState.sortRegistrationDate = null;
+		filterCards(usersData);
+  });
+  
+  buttonSortZA.addEventListener("click", ({ target }) => {
+		filtersState.sortName = target.id;
+		filtersState.sortAge = null;
+		filtersState.sortRegistrationDate = null;
+		filterCards(usersData);
+	});
+
+	buttonSort19.addEventListener("click", ({ target }) => {
+		filtersState.sortAge = target.id;
+		filtersState.sortName = null;
+		filtersState.sortRegistrationDate = null;
+		filterCards(usersData);
+  });
+  
+  buttonSort91.addEventListener("click", ({ target }) => {
+		filtersState.sortAge = target.id;
+		filtersState.sortName = null;
+		filtersState.sortRegistrationDate = null;
+		filterCards(usersData);
+	});
+
+	buttonSortOLD.addEventListener("click", ({ target }) => {
+		filtersState.sortRegistrationDate = target.id;
+		filtersState.sortAge = null;
+		filtersState.sortName = null;
+		filterCards(usersData);
+  });
+  
+  buttonSortNEW.addEventListener("click", ({ target }) => {
+		filtersState.sortRegistrationDate = target.id;
+		filtersState.sortAge = null;
+		filtersState.sortName = null;
+		filterCards(usersData);
+  });
+
+  buttonReset.addEventListener("click", () => {
```

Comment:
You forgot to reset the radio and range inputs in your UI

Reply:
✅ Done